cmake/ConfigureDoxygen.cmake

```diff
@@ -52,12 +69,31 @@ set(DOXYGEN_EXCLUDE_PATTERNS */CurrentlyExcluded/* */cmake/* */deploy/* */Docume
 
 configure_file(${DOXYFILE_CMAKE} ${DOXYFILE} @ONLY) 
 
+# add source code to doxygen input
+set(DOXYGEN_INPUT ${CMAKE_CURRENT_SOURCE_DIR})
+
 # Documentation for entire OPEN INFRA PLATFORM project.
+if(DOXYGEN_OPTIONAL_INCLUDE_EARLYBINDING)
+# add IFC generated files to doxygen input directory
+foreach(format ${IFC_FORMATS})	
+	if(EXISTS ${CMAKE_BINARY_DIR}/EarlyBinding/${format})
+		set(DOXYGEN_INPUT ${DOXYGEN_INPUT} "${CMAKE_BINARY_DIR}/EarlyBinding/${format}/src")
+	endif()
+endforeach()
 doxygen_add_docs(OpenInfraPlatform.GenerateDocumentation ALL
 ${DOXYGEN_INPUT} 	# Source code that is (to be) documented located here
 WORKING DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}	# Current root. Change if relative base point should be different.
+#USE_STAMP_FILE
+COMMENT "Generating doxymentation for TUM Open Infra Platform project.")
+else()
+doxygen_add_docs(OpenInfraPlatform.GenerateDocumentation ALL
+${CMAKE_CURRENT_SOURCE_DIR} 	# Source code that is (to be) documented located here
+WORKING DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}	# Current root. Change if relative base point should be different.
+#USE_STAMP_FILE
 COMMENT "Generating doxymentation for TUM Open Infra Platform project.")

```

Comment:
This is unnecessary. The DOXYGEN_INPUT variable gets set and optionally expanded by EarlyBinding stuff. Therefore, the endif(DOXYGEN_OPTIONAL_INCLUDE_EARLYBINDING) needs to come before doxygen_add_docs(..). If unclear, contact me.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/104/commits/1d8282d0f106b3c2598f552ed9c5d9e6434692e5